lib/services/collections_service.dart

```diff
@@ -308,18 +313,47 @@ class CollectionsService {
   }
 
   void _cacheCollectionAttributes(Collection collection) {
+    _collectionIDToCollections[collection.id] = collection;
```

Comment:
Isn't this a duplicate of L#321?

Reply:
i need to have the collection in the cache before calling `getCollectionDecryptedName` as it uses `getCollectionKey` which takes `collectionID` as parameter and finds the collection in the cache

Reply:
What if we restructure code to something like this:
```dart
  Uint8List getCollectionKey(int collectionID) {
    if (!_cachedKeys.containsKey(collectionID)) {
      final collection = _collectionIDToCollections[collectionID];
      _cachedKeys[collectionID] = _getDecryptedKey(collection);
    }
    return _cachedKeys[collectionID];
  }

  Uint8List _getDecryptedKey(Collection collection) {
    final encryptedKey = Sodium.base642bin(collection.encryptedKey);
    if (collection.owner.id == _config.getUserID()) {
      return CryptoUtil.decryptSync(encryptedKey, _config.getKey(),
          Sodium.base642bin(collection.keyDecryptionNonce));
    } else {
      return CryptoUtil.openSealSync(
          encryptedKey,
          Sodium.base642bin(_config.getKeyAttributes().publicKey),
          _config.getSecretKey());
    }
  }
```

Then you could call `_getDecryptedKey(...)`?

Reply:
Ya that's better